doc/mgmt/SONiC_Management_Framework_sFlow.md

```diff
@@ -0,0 +1,373 @@
+# High Level Design Document
+#### Rev 0.3
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/09/2019  |   Garrick He       | Initial version                   |
+| 0.2 | 10/04/2019  |   Garrick He       | Address review comments           |
+| 0.3 | 10/11/2019  |   Garrick He       | Address review comments           |
+| 0.4 | 10/15/2019  |   Garrick He       | Add information on default value  |
+|     |             |                    | and SONiC sFlow YANG              |
+
+# About this Manual
+This document provides general information about sFlow support in SONiC Management Framework
+# Scope
+This document describes the high level design of sFlow support in SONiC Management Framework. The underlying sFlow
+support in SONiC is provided by this high-level design document:
+https://github.com/Azure/SONiC/blob/master/doc/sflow/sflow_hld.md
+
+
+# 1 Feature Overview
+This feature will allow the user to configure sFlow using SONiC Management Framework with REST or gNMI. Translib will make changes
+to CONFIG DB to make configuration changes to sFlow. The underlying sFlow DB schema is already provided by sFlow support from SONiC
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+
+### 1.1.2 Configuration and Management Requirements
+1. CLI configuration/show support
+2. REST API support
+3. gNMI support
+
+### 1.1.3 Scalability Requirements
+
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+1. Implement sFlow support using transformer in sonic-mgmt-framework.
+
+### 1.2.2 Container
+There will be changes in the sonic-mgmt-framework container. The backend will be modifications to translib done through Transformer. There will be additional files added to:
+1. XML file for the CLI
+2. Python script to handle CLI request (actioner)
+3. Jinja template to render CLI output (renderer)
+4. sFlow YANG model
+
+### 1.2.3 SAI Overview
+
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+## 2.2 Functional Description
+
+
+# 3 Design
+## 3.1 Overview
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+This feature will allow the user to make/show sFlow configuration changes to CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+
+
+## 3.4 SyncD
+
+
+## 3.5 SAI
+
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+There are no OpenConfig YANG models available for sFlow so additions were made to SONiC YANG.
+Supported SONiC YANG URIs available from Swagger WebUI:
+```
```

Comment:
Do you need to update this section or just link to the actual code that has been merged?

Reply:
done.